lms/services/canvas.py

```diff
@@ -1,36 +1,79 @@
 from typing import Optional
 
 from lms import models
-from lms.services.exceptions import CanvasFileNotFoundInCourse
+from lms.services.exceptions import CanvasAPIPermissionError, CanvasFileNotFoundInCourse
 
 
 class CanvasService:
     """A high level Canvas service."""
 
     api = None
 
-    def __init__(self, canvas_api):
+    def __init__(self, canvas_api, file_service):
         self.api = canvas_api
-
-    def public_url_for_file(self, file_id, course_id, check_in_course=False):
-        """
-        Get a public URL for a Canvas file.
-
-        This will also attempt to check if the file is in the course to detect
-        course copy situations and fix it if we can.
-
-        :param file_id: The file to look up
-        :param course_id: The course the file should be in
-        :param check_in_course: Raise CanvasFileNotFoundInCourse if file_id isn't in
-            course_id
-        :return: A URL suitable for public presentation of the file
-        :raise  CanvasFileNotFoundInCourse: if check_in_course=True and file_id isn't in course_id
-            be in the course but isn't.
-        """
-
-        if check_in_course:
-            if not self.can_see_file_in_course(file_id, course_id):
-                raise CanvasFileNotFoundInCourse(file_id)
+        self._file_service = file_service
+
+    def public_url_for_file(
+        self, module_item_configuration, file_id, course_id, check_in_course=False
```

Comment:
`module_item_configuration` is probably better referred to as `assigment` throughout. It is the return value of the `assignment` service and is more concise and easier to read.

Reply:
I think we should just bite the bullet and rename `ModuleItemConfiguration` to `Assignment` everywhere. I started a separate PR to do that: https://github.com/hypothesis/lms/pull/2912